components/notebook-controller/Dockerfile

```diff
@@ -3,32 +3,60 @@
 # The Docker context is expected to be:
 #
 # ${PATH_TO_KUBEFLOW/KUBEFLOW repo}/components
-#
-# This is necessary because the Jupyter controller now depends on
-# components/common
+
+
+# Build arguments
+ARG SOURCE_CODE=.
 ARG GOLANG_VERSION=1.20
-FROM golang:${GOLANG_VERSION} as builder
 
+# Use ubi8/go-toolset as base image
+FROM registry.redhat.io/ubi8/go-toolset:${GOLANG_VERSION} as builder
```

Comment:
Are we okay to have a red hat specific image in an upstream build? :thinking:

Reply:
It's ubi8, I think we absolutely do want to use ubi in upstream! I'm more unsure about naming the user `rhods`, but that does not carry any bits with it, it's just strange identifier in context of opendatahub-io.

Reply:
So, yeah, we want to minimize upstream-downstream differences, so I say let's go for this. In the future, when we want to make downstream change, we can make it upstream first, check it in odh nightlies, and then put it into gitlab for downstream build.